sw/airborne/modules/nav/nav_rotorcraft_hybrid.c

```diff
@@ -118,13 +135,19 @@ static void nav_hybrid_goto(struct EnuCoor_f *wp)
     float max_speed_decel2 = fabsf(2.f * dist_to_wp * nav_max_deceleration_sp); // dist_to_wp can only be positive, but just in case
     float max_speed_decel = sqrtf(max_speed_decel2);
     // Bound the setpoint velocity vector
-    max_h_speed = Min(nav_goto_max_speed, max_speed_decel); // use hover max speed
```

Comment:
I think this will break the functionality of nav_goto_max_speed, for those users that expect it to work.

Reply:
I have checked in master who uses this nav_goto_max_speed and actually nobody uses it. Only us (Rotatingwing) "mis"-use it the state machine assigning it the sane value as nav_max_speed. I would remove it completly?

Reply:
I have added this because our hybrid had issues when doing small moves. So it assumes that a goto is short move that should be done close to the hover attitude, while a segment is for fast forward flights.

Reply:
I  would prefer if this was automatic. If a hybrid drone type has a distance that is too short to transition (partially) and should stay in hover mode while moving the small distance, then define
```NAV_HYBIRD_MIN_DISTANCE_TO_TRANSISITON```

```
#if NAV_HYBRID_MIN_DISTANCE_TO_TRANSISTION
  static bool may_transisition = false;
  if (wp_dist > NAV_HYBRID_MIN_DISTANCE_TO_TRANSISTION) {
    may_transition = true;
  }
  if ((wp_dist < NAV_HYBRID_MIN_DISTANCE_TO_TRANSISTION) && (may_transisition == false)) {
    max_h_speed = Min(nav_goto_max_speed, max_speed_decel);
  }
  if (max_speed_decel < nav_goto_max_speed) {
    may_transition = false;
  }
#endif
```

@gautierhattenberger what do you think?

Reply:
I agree that going for a transition should be automatic for "normal" use, and setting a distance threshold is not a bad option to start with.
Still, I can add to my previous comment that I was using this goto=slow speed, route = high speed as a "debug" feature. If the forward mode doesn't really work, I can still navigate back to home or standby like a regular rotorcraft. It is not an ideal situation, and probably not a good "default" behavior, but I think it is worth keeping this as an option for now (e.g. a config flag should be available to force this behavior if needed).

Reply:
@gautierhattenberger I have added a definition that allows you to force max goto speed so that you can use the goto as a "debug". Let me know if it is ok with you.

Reply:
From my perspective, this is not a general difference between goto and route: for instance flying back home due to an exception I would do with a goto, but this can be from far away.
With NAV_HYBIRD_MIN_DISTANCE_TO_TRANSISITON I see the problem that the transition should be a function of airspeed, and a transition to forward flight may already be needed in order to hover. On the other hand, I can imagine that for vehicles that can fly very fast in hover and have an airspeed range in which they cannot fly (read: they need some dynamic transition maneuver) this can make sense. Not sure if we have any of those vehicles in Paparazzi though.

Personally I would vote for remove this difference between route and goto for these reasons, but if it is useful for some I'm ok with keeping it.

Reply:
I agree that the question is not about goto or route, nor preventing a "transition" if it is required to hover due to wind condition. But sometimes it is required to set the desired and/or maximum speed, independently from the type . This should be done from the flight plan in the first place I think. At the moment, the different parameters must be changed each time before calling a flight instruction, which is a pain. I guess enhancing the flight plans is required at longer (short ?) term.
In the meantime, I just would like to preserve a "debug" mode where go and route have different maximum ground speed, even if it is very basic. Why the current state where `goto_max_speed` is equal to `max_speed` unless specified otherwise is a problem ?
Sorry if I'm a bit lost, but I'm not sure what was the original issue here...

Reply:
I agree it is a pain to do it after each flight plan instruction, but if it is just for debugging it is maybe not a big problem? Just thinking that the more simple the code is the easier it is to understand and maintain. If you still disagree we can stick to the goto_max_speed implementation.

Reply:
Ok, let's keep the code simple.